core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -291,10 +291,10 @@ static TableMetadata fromJson(FileIO io, InputFile file, JsonNode node) {
           schema, TableMetadata.INITIAL_SPEC_ID, node.get(PARTITION_SPEC)));
     }
 
-    Integer lastAssignedPartitionId = JsonUtil.getIntOrNull(LAST_ASSIGNED_PARTITION_ID, node);
+    Integer lastAssignedPartitionId = JsonUtil.getIntOrNull(LAST_PARTITION_ID, node);
```

Comment:
Won't this break backwards compatibility?

Reply:
This property was just added, so the rename is safe.